src/StakeVault.sol

```diff
@@ -51,6 +52,20 @@ contract StakeVault is Ownable {
         stakeManager = _stakeManager;
     }
 
+    /**
+     * @notice Registers the vault with the stake manager.
+     */
+    function register() external {
+        stakeManager.registerVault();
+    }
+
+    /**
+     * @notice Returns the address of the current owner.
+     */
+    function owner() public view override(Ownable, IStakeVault) returns (address) {
+        return super.owner();
+    }
```

Comment:
In `StakeManager.registerVault()` we're reading `StakeVault.owner()` so I added `owner()` to the `IStakeVault` interface.

Needed to explicitly add `owner()` here to make this compile.